Clarify hash-to-curve properties at a high level. (#319)
cfrg/draft-irtf-cfrg-hash-to-curve: draft-irtf-cfrg-hash-to-curve.md

Many cryptographic protocols require a procedure that encodes an arbitrary input,
e.g., a password, to a point on an elliptic curve. This procedure is known
as hashing to an elliptic curve, where the hashing procedure provides collision

> resistance over the set of inputs and produces an output point for which the

Reviewer: `producing a point for which the DL is difficult to find` is not a property of the hash-to-curve function, it is a property of the ccurve itself (or the (sub)group of points).

Better to say that one application of this hash function is for generating points on the curve such that _the caller_ cannot know its discrete log.

Author: Your proposed text seems to match what's written here?

Author: (Chatted offline and applied a change)

> resistance over the set of inputs and produces an output point for which the
> discrete log is unknown. Prominent examples of cryptosystems that hash to elliptic curves include

Reviewer 2: ```suggestion
resistance and does not reveal the discrete log relationship between the
output point and any distinguished point on the curve (e.g., a generator point).
Prominent examples of cryptosystems that hash to elliptic curves include
```

I don't 100% love the wording I'm proposing here, but the rationale for the suggested edit:

- this sentence formerly referred to a set of inputs; but the first sentence says the inputs are arbitrary, which isn't quite in harmony
- discrete log is a relation between two points, so it doesn't really make sense to say the discrete log of a point is unknown without specifying the base. Being super persnickety: the discrete log of a point to itself is always known!

Suggestions for improvement? I'll keep thinking about this, too.

Author: I imagine we can bike shed on this endlessly. Honestly, I don't think we need to be super precise here, as this is in the intro and isn't (IMO) wrong. I'd accept this change as-is and move on.

Reviewer: I like the first part;
for the second part, it's true that talking about the discrete log one must specify a base point, however I think this does not need to be emphasized here. 

`resistance and does not reveal the discrete logarithm of the output point. Prominent examples of algorithms that hash to elliptic curves include`

Reviewer 2: OK, you've both convinced me that it's OK to be slightly imprecise here :+1:

Reviewer 2: Making this concrete:

```suggestion
resistance and does not reveal the discrete log of the output point.
Prominent examples of cryptosystems that hash to elliptic curves include
```

> discrete log is infeasible to compute. Prominent examples of cryptosystems that hash to elliptic curves include

Author: ```suggestion
discrete log is unknown. Prominent examples of cryptosystems that hash to elliptic curves include
```

Many cryptographic protocols require a procedure that encodes an arbitrary input,
e.g., a password, to a point on an elliptic curve. This procedure is known
as hashing to an elliptic curve, where the hashing procedure provides collision

> resistance and does not reveal the discrete log of the output point.

Reviewer: ```suggestion
resistance and does not reveal the discrete logarithm of the output point.
```
